contracts/SwapperRegistry.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.3;
+
+import '@openzeppelin/contracts/utils/structs/EnumerableSet.sol';
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+
+interface ISwapperRegistry {
+  event SwapperAdded();
+  event SwapperNameUpdated();
+  event SwapperAddressUpdated();
+  event SwapperRemoved();
+  event SwapperAndTokenEnabled();
+}
+
+contract SwapperRegistry is ISwapperRegistry {
+  // Add the library methods
+  using SafeERC20 for IERC20;
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  mapping(address => string) public nameByAddress;
+  mapping(string => address) public swapperByName;
+  mapping(address => uint256) public initializationByAddress;
+  EnumerableSet.AddressSet internal _swappers;
+
+  mapping(address => EnumerableSet.AddressSet) internal _approvedTokensBySwappers;
+
+  constructor() {
+    // TODO: set governance
+    // TODO: set collectable dust
```

Comment:
You can use `UtilsReady` abstract contract from library

Reply:
Can't it's in old pragma :sob: will probably import it and update the pragma.